Kosystem/Services/KoInMemoryRepository.cs

```diff
@@ -0,0 +1,182 @@
+        {
+            return _people.TryRemove(personId, out _);
+        }
+
+        public bool DeleteRoom(int roomId)
+        {
+            var peopleInRoom = FindPeopleInRoom(roomId);
+
+            foreach (var person in peopleInRoom)
+            {
+                _people.TryUpdate(person.Id, person.WithoutRoom(), person);
+            }
+
+            _peoplePerRoom.TryRemove(roomId, out _);
+            return _people.TryRemove(roomId, out _);
+        }
+
+        public bool DequeuePerson(int personId)
+        {
+            var person = FindPerson(personId);
+            if (person.HasValue && person.Value.IsEnqueued)
+            {
+                return _people.TryUpdate(personId, person.Value.AsDequeued(), person);
+            }
+
+            return false;
+        }
+
+        public bool EnqueuePerson(int personId)
+        {
+            var person = FindPerson(personId);
+            if (person.HasValue && !person.Value.IsEnqueued)
+            {
+                return _people.TryUpdate(personId, person.Value.AsEnqueued(DateTime.Now), person);
+            }
+
+            return false;
+        }
+
+        public IReadOnlyCollection<PersonModel> FindPeopleInRoom(int roomId)
+        {
+            if (_peoplePerRoom.TryGetValue(roomId, out var peopleInRoom))
+            {
+                var list = new List<PersonModel>(peopleInRoom.Count);
+
+                foreach (var personId in peopleInRoom)
+                {
+                    var person = FindPerson(personId);
+                    if (person.HasValue)
+                    {
+                        list.Add(person.Value);
+                    }
+                }
+
+                return list;
+            }
+
+            return Array.Empty<PersonModel>();
+        }
+
+        public PersonModel? FindPerson(int personId)
+        {
+            return _people.TryGetValue(personId, out var person) ? person : null;
+        }
+
+        public RoomModel? FindRoom(int roomId)
+        {
+            return _rooms.TryGetValue(roomId, out var room) ? room : null;
+        }
+
+        public bool RemovePersonFromRoom(int roomId, int personId)
+        {
+            if (_peoplePerRoom.TryGetValue(roomId, out var peopleInRoom))
+            {
+                return peopleInRoom.Remove(personId);
+            }
+
+            return false;
+        }
+
+        public PersonModel? UpdatePerson(UpdatePersonModel updatePerson)
+        {
+            var person = FindPerson(updatePerson.Id);
+            if (!person.HasValue)
+            {
+                return null;
+            }
+
+            PersonModel updatedPerson = person.Value.Update(updatePerson);
+            if (_people.TryUpdate(person.Value.Id, updatedPerson, person))
+            {
+                return updatedPerson;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public RoomModel? UpdateRoom(UpdateRoomModel updateRoom)
```

Comment:
Codacy found an issue: [Rename the parameter 'updateRoom' so that it does not duplicate the method name.](https://app.codacy.com/gh/HelloWorldSweden/Kosystem/pullRequest?prid=6010716)